src/components/material/MaterialHeader.tsx

```diff
@@ -78,6 +80,34 @@ const MaterialHeader: React.FC<MaterialHeaderProps> = ({
 
   const title = containsDanish ? fullTitle : `${fullTitle} (${allLanguages})`;
   const coverPid = pid || getManifestationPid(manifestations);
+  const { track } = useStatistics();
+
+  // This is used to track whether the user is changing between manterial types or just clicking the same button over
+  const [manifestationMaterialTypes, setManifestationMaterialTypes] = useState<
+    string[]
+  >([]);
+
+  useEffect(() => {
+    const newManifestationMaterialTypes = manifestation.materialTypes.map(
+      (item) => item.specific
+    );
+    if (isEqual(manifestationMaterialTypes, newManifestationMaterialTypes)) {
+      return;
+    }
```

Comment:
Consider creating the list of material types outside of the hook and then using the list as a dependency of the hook.

This should make the code less complex.

Also note that you may need `useDeepCompareEffect()` here.

Reply:
That makes total sense. Thanks! :)